Hazel/src/Hazel/Application.cpp

```diff
@@ -78,4 +79,10 @@ namespace Hazel {
 		return true;
 	}
 
-}
\ No newline at end of file
+	bool Application::OnWindowResize(WindowResizeEvent& e)
+	{
+		m_Window->GetContext()->UpdateViewport(e.GetWidth(), e.GetHeight());
+		return false; // Allow layers to get this event
```

Comment:
It's because of how the EventDispatcher works.

Reply:
EventDispatcher only sets Handled to true or  false

Reply:
The intent with having methods responding to events is so that each layer can handle the event if it chooses, and then decide if the event should still be passed down to lower layers. Thus we make the return value false here so that the event does get passed down to lower layers.
The mechanism for this is the setting of Handled which is then checked in Application's OnEvent function when it calls the layers' OnEvent functions.

Reply:
In `Application::OnEvent(...)` we dispatch the event first to application itself. Returning true in the event function will set `Event::Handled` to true. When iterating to the layerstack, after the first layer, it will be set to true and break out, resulting in distributing the event only to 1 layer, independed if it is handled or not.

We should add an if check before entering the for loop, in case application handled it (for example the close event)

Reply:
Alternatively we could have the loop check before propagating the event, instead of propagating before checking. This also avoids the check after the last layer which is unnecessary.

Reply:
I was about to write that, it will look nicer yes, i was thinking of the iterator incrementing first, but we can indeed fix this.

Reply:
I think if it's nicer for the "end-programmer" to use then I wouldn't consider it micro optimization and more so a better experience for the developer. But I think that's for another time.

Reply:
Create an issue for it in the meantime, as it is a bug that it passes close events to the layers... it should never go into the for loop to begin with...

Reply:
Other than that would this be available for merge?

Reply:
I first have to have time to test it fully... probably next week (as my pc will have downtime due to yearly maintenance comming up)

Reply:
I'd be glad to test what is needed if you can specify what needs to be tested.

Reply:
- test switching from each mode to each mode
- test if each event is fired correctly

Reply:
Can confirm you can switch between modes.